Desorption added to PorousFlow

WilkAndy opened the pull request:
Added: time derivative Kernels for the desorped fluid mass.
Added: dependence of PorousFlow on ChemicalReactions
Small tidy-up to the Postprocessors

Fixes #8240
Refs #8139

WilkAndy committed:
Code

Added: time derivative Kernels for the desorped fluid mass.
Added: dependence of PorousFlow on ChemicalReactions
Small tidy-up to the Postprocessors

Fixes #8240

WilkAndy committed:
Tests and doco of PorousFlow desorption.

Fixes #8240
Refs #8139

moosebuild commented:
Job `Pthreads test:linux-gnu` on f126470 : invalidated by @permcody

time_steppers/time_adaptive.test timeout

permcody commented on modules/porous_flow/Makefile:
```diff
@@ -17,9 +17,10 @@ FRAMEWORK_DIR      ?= $(MOOSE_DIR)/framework
+# dependency on tensor_mechanics and fluid_properties and chemical_reactions
 TENSOR_MECHANICS := yes
 FLUID_PROPERTIES := yes
+CHEMICAL_REACTIONS := yes
```
Whoa! This will require a change to modules.mk too. This is for the case where somebody uses only PourousFlow from yet another application (Maybe Ba-hun) but doesn't specify "chemical reactions".

WilkAndy replied:
Aha, thanks @permcody , i'll look into modules.mk today.

permcody reviewed (changes requested):
We need the associated change in modules.mk too before this can be merged. It'll be more than a single line because that file has some order dependency which may have to be adjusted to accommodate this change.

WilkAndy committed:
makefile changes

Originally i had chemical_reactions:=yes in moose/unit/Makefile, but now i don't think that's necessary
Hopeully moose/modules/modules.mk now has chemical_reactions inserted in all the right places

Fixes #8240

permcody commented:
Superseded by #8305
